src/Microsoft.OData.Edm/Schema/CoreModel/EdmCoreModel.cs

```diff
@@ -7,123 +7,119 @@
+                new EdmCoreModelPrimitiveType("Date", EdmPrimitiveTypeKind.Date),
+                new EdmCoreModelPrimitiveType("Decimal", EdmPrimitiveTypeKind.Decimal),
+                new EdmCoreModelPrimitiveType("Binary", EdmPrimitiveTypeKind.Binary),
+                new EdmCoreModelPrimitiveType("String", EdmPrimitiveTypeKind.String),
+                new EdmCoreModelPrimitiveType("Stream", EdmPrimitiveTypeKind.Stream),
+
+                new EdmCoreModelPrimitiveType("Geography", EdmPrimitiveTypeKind.Geography),
+                new EdmCoreModelPrimitiveType("GeographyPoint", EdmPrimitiveTypeKind.GeographyPoint),
+                new EdmCoreModelPrimitiveType("GeographyLineString", EdmPrimitiveTypeKind.GeographyLineString),
+                new EdmCoreModelPrimitiveType("GeographyPolygon", EdmPrimitiveTypeKind.GeographyPolygon),
+                new EdmCoreModelPrimitiveType("GeographyCollection", EdmPrimitiveTypeKind.GeographyCollection),
+                new EdmCoreModelPrimitiveType("GeographyMultiPolygon", EdmPrimitiveTypeKind.GeographyMultiPolygon),
+                new EdmCoreModelPrimitiveType("GeographyMultiLineString", EdmPrimitiveTypeKind.GeographyMultiLineString),
+                new EdmCoreModelPrimitiveType("GeographyMultiPoint", EdmPrimitiveTypeKind.GeographyMultiPoint),
+                new EdmCoreModelPrimitiveType("Geometry", EdmPrimitiveTypeKind.Geometry),
+                new EdmCoreModelPrimitiveType("GeometryPoint", EdmPrimitiveTypeKind.GeometryPoint),
+                new EdmCoreModelPrimitiveType("GeometryLineString", EdmPrimitiveTypeKind.GeometryLineString),
+                new EdmCoreModelPrimitiveType("GeometryPolygon", EdmPrimitiveTypeKind.GeometryPolygon),
+                new EdmCoreModelPrimitiveType("GeometryCollection", EdmPrimitiveTypeKind.GeometryCollection),
+                new EdmCoreModelPrimitiveType("GeometryMultiPolygon", EdmPrimitiveTypeKind.GeometryMultiPolygon),
+                new EdmCoreModelPrimitiveType("GeometryMultiLineString", EdmPrimitiveTypeKind.GeometryMultiLineString),
+                new EdmCoreModelPrimitiveType("GeometryMultiPoint", EdmPrimitiveTypeKind.GeometryMultiPoint),
+
+                new EdmCoreModelPrimitiveType("PrimitiveType", EdmPrimitiveTypeKind.PrimitiveType), // Edm.PrimitiveType
+            };
+
+            foreach (var primitive in primitiveTypes)
+            {
+                _primitiveTypeKinds[primitive.Name] = primitive.PrimitiveKind;
+                _primitiveTypeKinds[primitive.Namespace + '.' + primitive.Name] = primitive.PrimitiveKind;
+                 _primitiveTypesByKind[primitive.PrimitiveKind] = primitive;
+
+                _coreSchemaTypes[primitive.Namespace + '.' + primitive.Name] = primitive;
+                _coreSchemaTypes[primitive.Name] = primitive;
+
+                _coreSchemaElements.Add(primitive);
+            }
+
+            _coreSchemaElements.Add(_complexType); // Edm.EntityType

```

Comment:
>// Edm.EntityType [](start = 51, length = 17)

I think you switched the comment for Edm.ComplexType and Edm.EntityType

Reply:
@Resolved.